Unable to build module without verbs with a subpackage that has verbs

Summary:
- Create a module, but don't define any verbs in it
- Create a subpackage in that module and define a verb
- Building the module will fail because `main.go` imports the module package but then doesn't use it

Detailed example:

<img width="167" alt="Screenshot 2024-07-24 at 11 44 34 AM" src="https://github.com/user-attachments/assets/914c5708-6cfc-4e02-9807-1c4e0445c795">

`outer.go`:

```go
package outer

import "ftl/outer/inner"

//ftl:data export
type Event struct {
	Value inner.EchoResponse
}

```
`inner.go`:
```go
package inner

import (
	"context"
	"fmt"

	"github.com/TBD54566975/ftl/go-runtime/ftl" // Import the FTL SDK.
)

//ftl:data export
type EchoRequest struct {
	Name ftl.Option[string] `json:"name"`
}

//ftl:data export
type EchoResponse struct {
	Message string `json:"message"`
}

//ftl:verb export
func Echo(ctx context.Context, req EchoRequest) (EchoResponse, error) {
	return EchoResponse{Message: fmt.Sprintf("Hello, %s!", req.Name.Default("anonymous"))}, nil
}
```

Building this module generates `.ftl/go/main/main.go`:
```go
// Code generated by FTL. DO NOT EDIT.
package main

import (
	"context"

	"github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/ftlv1connect"
	"github.com/TBD54566975/ftl/common/plugin"
	"github.com/TBD54566975/ftl/go-runtime/server"

	"ftl/outer"
	"ftl/outer/inner"
)

func main() {
	verbConstructor := server.NewUserVerbServer("outer",
		server.HandleCall(inner.Echo),
	)
	plugin.Start(context.Background(), "outer", verbConstructor, ftlv1connect.VerbServiceName, ftlv1connect.NewVerbServiceHandler)
}
```

Which results in the error:
`./main.go:11:2: "ftl/outer" imported and not used`

--- a/go-runtime/compile/build.go
+++ b/go-runtime/compile/build.go
@@ -442,9 +442,6 @@ var scaffoldFuncs = scaffolder.FuncMap{
 	},
 	"mainImports": func(ctx mainModuleContext) []string {
 		imports := sets.NewSet[string]()
-		if len(ctx.Verbs) > 0 {
-			imports.Add(ctx.Name)
-		}
 		for _, v := range ctx.Verbs {
 			imports.Add(strings.TrimPrefix(v.MustImport, "ftl/"))
 		}

